vars/getBuildNumber.groovy

```diff
@@ -0,0 +1,7 @@
+/**
+  * Retrieves formatted official build number.
+  */
+def call() {
+    def versionNumber = VersionNumber('${BUILD_YEAR}${BUILD_MONTH, XX}${BUILD_DAY, XX}-${BUILDS_TODAY, XX}')
```

Comment:
For debugging, can you put asserts in here to ensure that the environment variables you are depending on are installed (along with a note of the required plugin upon failed assertion)?